core/src/main/scala/cats/instances/seq.scala

```diff
@@ -134,7 +133,7 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
         G.map(Chain.traverseViaChain(fa.toIndexedSeq)(f))(_.toVector)
 
       override def mapWithIndex[A, B](fa: Seq[A])(f: (A, Int) => B): Seq[B] =
-        fa.iterator.zipWithIndex.map(ai => f(ai._1, ai._2)).toIndexedSeq
+        fa.zipWithIndex.map(ai => f(ai._1, ai._2))
```

Comment:
Apologies if I missed it, why the change here?

Reply:
Check my last comment:

> replaced a previous usage of toIndexedSeq on mapWithIndex with a version that does two traversals but preserves the underlying class

Motivated by

> Also I still think that it would be better to preserve the original Seq's concrete type

Reply:
FWIW I think we can use .view here to get the best of both worlds, but anyway.

Reply:
@armanbilge how would we get back the original collection using `view`?

Reply:
> (Since version 2.13.0) Views no longer know about their underlying collection type; .force always returns an IndexedSeq

https://www.scala-lang.org/api/current/scala/collection/View.html#force:scala.collection.IndexedSeq[A]

Reply:
Aha, casualty of the new collections, my bad :)

In 2.13 though you can do this:
```scala
fa.view.zipWithIndex.map(ai => f(ai._1, ai._2)).to(fa.factory)
```

But, too much shenanigans. I'm also skeptical about instances for `Seq`.